src/main/java/org/marketdesignresearch/mechlib/outcomerules/ccg/constraintgeneration/PerBidConstraintGenerator.java

```diff
@@ -31,14 +33,15 @@
  */
 class PerBidConstraintGenerator implements PartialConstraintGenerator {
     @Override
-    public void generateFirstRoundConstraints(BundleValueBids<?> bids, Outcome referencePoint, Map<Good, PotentialCoalition> goodToBidderMap, CorePaymentRule corePaymentRule) {
+    public void generateFirstRoundConstraints(BundleValueBids<?> bids, Outcome referencePoint, Map<Good, Set<PotentialCoalition>> goodToBidderMap, CorePaymentRule corePaymentRule) {
         corePaymentRule.resetResult();
         Allocation allocation = referencePoint.getAllocation();
         Payment lowerBound = referencePoint.getPayment();
         for (Entry<Bidder, ? extends BundleValueBid<?>> bid : bids.getBidMap().entrySet()) {
             for (BundleExactValuePair bundleBid : bid.getValue().getBundleBids()) {
-                // TODO: assumes availability of 1
-                Set<Bidder> blockedBiddersSet = bundleBid.getBundle().getBundleEntries().stream().map(BundleEntry::getGood).filter(goodToBidderMap::containsKey).map(good -> goodToBidderMap.get(good).getBidder()).collect(Collectors.toSet());
+                // TODO check implementation for availability of more than 1
+            	
+                Set<Bidder> blockedBiddersSet = bundleBid.getBundle().getBundleEntries().stream().map(BundleEntry::getGood).filter(goodToBidderMap::containsKey).flatMap(good -> goodToBidderMap.get(good).stream().map(PotentialCoalition::getBidder)).collect(Collectors.toSet());
```

Comment:
I added a test for this fix here and it works in very simple worlds. I did not have time to dive into the details of this constraint generator and do not know if the fix generalizes to very hard worlds.

Reply:
Maybe a test with an easy, but multi-availability world (similar to what I've written) would be enough? I think it should mostly generalize to more complex worlds if the support for one multi-availability item is give. But it's rather a gut feeling, it's even better to have a more complex world tested as well.

Reply:
This test is already there. From my understanding at the moment this should be fine however. The PerBidConstrainGenerator only adds some lower bounds based on bids but is not sufficient to generate all required constraint. Therefore it needs to be combined with another ConstrainGenerator.

Reply:
ok, great! that should be sufficient for now, then.